pytest-embedded-qemu/tests/test_qemu.py

```diff
@@ -33,6 +33,32 @@ def test_pexpect_by_qemu(dut):
     result.assert_outcomes(passed=1)
 
 
+@qemu_bin_required
+def test_pexpect_make_restart_by_qemu_xtensa(testdir):
+    testdir.makepyfile("""
+        import pexpect
+        import pytest
+
+        def test_pexpect_by_qemu(dut):
+            dut.expect('Hello world!')
+            dut.hard_reset()
```

Comment:
The test case failed locally.

Seems like the `hard_reset` function is missing in the `IdfUnityDutMixin`.

For now let's add `hard_reset` in the `QemuDut` as a temporary workaround, and improve the dependency chain later.

Reply:
Let's do it